brainiak/utils/fmrisim.py

```diff
@@ -469,6 +469,16 @@ def generate_stimfunction(onsets,
         # Pull out the onsets, weights and durations, set as a float
         for line in text:
             onset, duration, weight = line.strip().split()
+
+            # Check if the onset is more precise than the temporal resolution
+            dp = len(onset[onset.find('.') + 1:])
```

Comment:
There may be issue if some digits are written in scientific format, e.g., 1.0234e+03. We may want to comment in the docstring that such formatting can cause problem. Or, maybe we can solve it by converting the string to float and find the resolution in floating number.

Reply:
Wow astute observation! This was a great comment and I have hopefully fixed it now